TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs

```diff
@@ -372,5 +372,113 @@ public LockTableStatement CreateLockTableStatement(CodeElementsParser.LockTableS
             var savepointName = context.Diagnostics == null ? new SymbolReference(new AlphanumericValue((Token)context.savepoint_name), SymbolType.SqlIdentifier) : null;
             return new ReleaseSavepointStatement(savepointName);
         }
+
+        public SetAssignmentStatement CreateSetAssignmentStatement(CodeElementsParser.SetAssignmentStatementContext context)
+        {
+            IList<Assignment> assignments = context.assignmentClause().Select(CreateAssignmentClause).ToList();
+            return new SetAssignmentStatement(assignments);
+        }
+
+        private Assignment CreateAssignmentClause(CodeElementsParser.AssignmentClauseContext context)
+        {
+            if (context.simpleAssignmentClause() != null)
+            {
+                return CreateSimpleAssignmentClause(context.simpleAssignmentClause());
+            }
+            else if (context.multipleAssignmentClause() != null)
+            { 
+                return CreateMultipleAssignmentClause(context.multipleAssignmentClause());
+            }
+            return null;
+        }
+
+        private Assignment CreateMultipleAssignmentClause(CodeElementsParser.MultipleAssignmentClauseContext context)
+        {
+            IList<SourceValue> values = null;
```

Comment:
To ensure value list (and target list) is always non-null in an Assignment instance, you should initialize here to an empty list and call Add in the second if branch